### PR TITLE
Manged disk port from Anu (Samples/UTs included)

### PR DESCRIPTION
**Pull request by anudeepsharma**
Session record will come later.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anudeepsharma__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ans). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVHD/Program.cs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+                //=============================================================
+                // Deleting the custom image
+                Utilities.Log("Deleting custom Image: " + virtualMachineCustomImage.Id);
+
+                azure.VirtualMachineCustomImages.DeleteById(virtualMachineCustomImage.Id);
+
+                Utilities.Log("Deleted custom image");
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVHD/Program.cs]: you should not catch in the RunSample

File: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVM/Program.cs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                //=============================================================
+                // Deleting the custom image
+                Utilities.Log("Deleting custom Image: " + virtualMachineCustomImage.Id);
+
+                azure.VirtualMachineCustomImages.DeleteById(virtualMachineCustomImage.Id);
+
+                Utilities.Log("Deleted custom image");
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVM/Program.cs]: Same. No catch in RunSample

**Comment by hovsepm** [comment]
> using System.Reflection;

License header. 

---
Refers to: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVHD/Properties/AssemblyInfo.cs:1 in 2bd4ed9. [](commit_id = 2bd4ed96f84f743026008fd9aa0a9b738aa5615a, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header. 

---
Refers to: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingCustomImageFromVM/Properties/AssemblyInfo.cs:1 in 2bd4ed9. [](commit_id = 2bd4ed96f84f743026008fd9aa0a9b738aa5615a, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header. 

---
Refers to: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromSnapshot/Properties/AssemblyInfo.cs:1 in 2bd4ed9. [](commit_id = 2bd4ed96f84f743026008fd9aa0a9b738aa5615a, deletion_comment = False)

File: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromVhd/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromVhd/Properties/AssemblyInfo.cs]: License header.

**Comment by hovsepm** [comment]
> using System.Reflection;

License header. 

---
Refers to: Samples/ResourceManagement/Compute/ManageVirtualMachineWithDisk/Properties/AssemblyInfo.cs:1 in 2bd4ed9. [](commit_id = 2bd4ed96f84f743026008fd9aa0a9b738aa5615a, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header. 

---
Refers to: Samples/ResourceManagement/Compute/MigrateVirtualMachine/Properties/AssemblyInfo.cs:1 in 2bd4ed9. [](commit_id = 2bd4ed96f84f743026008fd9aa0a9b738aa5615a, deletion_comment = False)

File: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromSnapshot/Program.cs [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+                // Data disks SAS Uri
+                foreach (var disk in linuxVM2.DataDisks.Values)
+                {
+                    var dataDisk = azure.Disks.GetById(disk.Id);
+                    var dataDiskSasUri = dataDisk.GrantAccess(24 * 60);
+                    Utilities.Log($"Data disk (lun: {disk.Lun}) SAS Uri: {dataDiskSasUri}");
+                }
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromSnapshot/Program.cs]: No catch in RunSample

File: Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromVhd/Program.cs [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+                foreach (String diskId  in  dataDiskIds)
+                {
+                    var dataDisk = azure.Disks.GetById(diskId);
+                    var dataDiskSasUri = dataDisk.GrantAccess(24 * 60);
+                    Utilities.Log(String.Format("Data disk SAS Uri: %s", dataDiskSasUri));
+                }
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/CreateVirtualMachineUsingSpecializedDiskFromVhd/Program.cs]: No catch in RunSample

File: Samples/ResourceManagement/Compute/ManageVirtualMachineWithDisk/Program.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+                Console.WriteLine("Starting Linux VM");
+
+                linuxVM.Start();
+
+                Console.WriteLine("Started Linux VM");
+                Utilities.PrintVirtualMachine(linuxVM);
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/ManageVirtualMachineWithDisk/Program.cs]: No catch in RunSample

File: Samples/ResourceManagement/Compute/MigrateVirtualMachine/Program.cs [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                Utilities.Log("Migrate VM: " + linuxVM.Id);
+
+                linuxVM.MigrateToManaged();
+
+                Utilities.Log("Migrated VM: " + linuxVM.Id);
+
+                Utilities.PrintVirtualMachine(linuxVM);
+            }
+            catch (Exception f)
+            {
```
Comment (hovsepm) [inline Samples/ResourceManagement/Compute/MigrateVirtualMachine/Program.cs]: No catch in RunSample

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/Common/TestHelper.cs [hunk 8]
```diff
@@ -298,5 +298,42 @@ private static bool HandlerContains<T1>(DelegatingHandler handler)
...
                 && HandlerContains<T1>(handler.InnerHandler as DelegatingHandler)));
         }
+
+        public static void DeprovisionAgentInLinuxVM(string host, int port, string userName, string password)
+        {
+            if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                return;
+            }
+            //SSHShell shell = null;
```
Comment (hovsepm) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/Common/TestHelper.cs]: Why is this commented? Please add a text with explanation or remove.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs [hunk 9]
```diff
@@ -97,5 +105,24 @@ public AvailabilitySetImpl WithUpdateDomainCount(int updateDomainCount)
             Inner.PlatformUpdateDomainCount = updateDomainCount;
             return this;
         }
+
+        ///GENMHASH:F792F6C8C594AA68FA7A0FCA92F55B55:76F4D314E5BB1E6DEE9BFF0081B150DC
+        public AvailabilitySetSkuTypes Sku()
+        {
+            if (this.Inner.Sku != null && this.Inner.Sku.Name != null) {
+                return AvailabilitySetSkuTypes.Parse(this.Inner.Sku.Name);
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs]: curly bracket should go on on new line

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs [hunk 10]
```diff
@@ -97,5 +105,24 @@ public AvailabilitySetImpl WithUpdateDomainCount(int updateDomainCount)
...
+            }
+            return null;
+        }
+
+        ///GENMHASH:53AB73C440C52ADA0E332540DA0BEEB4:F6990CB6AF41DB826958D5810A250621
+        public AvailabilitySetImpl WithSku(AvailabilitySetSkuTypes skuType)
+        {
+            if (this.Inner.Sku == null) {
+                this.Inner.Sku = new Sku();
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs]: curly bracket should go on on new line

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/DataDiskImpl.cs [hunk 11]
```diff
@@ -1,254 +1,254 @@
...
-        {
-            return Parent;
-        }
-    }
-}
+//// Copyright (c) Microsoft Corporation. All rights reserved.
+//// Licensed under the MIT License. See License.txt in the project root for license information.
+
+//namespace Microsoft.Azure.Management.Compute.Fluent
+//{
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/DataDiskImpl.cs]: Please add a description why all the code is commented in this file.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/DiskImpl.cs [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+                return new DiskSkuTypes(this.Inner.AccountType.Value);
+            }
+            return null;
+        }
+
+        ///GENMHASH:0202A00A1DCF248D2647DBDBEF2CA865:4862DE76074C3C17570C425395A8E68C
+        public override async Task<Microsoft.Azure.Management.Compute.Fluent.IDisk> CreateResourceAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var diskInner = await client.CreateOrUpdateAsync(ResourceGroupName, Name, Inner);
+            SetInner(diskInner);
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/DiskImpl.cs]: you should pass cancellationToken to Async method calls

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/Domain/InterfaceImpl/VirtualMachineCustomImagesImpl.cs [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        /// <summary>
+        /// Asynchronously delete a resource from Azure, identifying it by its name and its resource group.
+        /// </summary>
+        /// <param name="groupName">The group the resource is part of.</param>
+        /// <param name="name">The name of the resource.</param>
+        /// <return>A completable indicates completion or exception of the request.</return>
+        async Task Microsoft.Azure.Management.Resource.Fluent.Core.CollectionActions.ISupportsDeletingByGroup.DeleteByGroupAsync(string groupName, string name, CancellationToken cancellationToken)
+        {
+            await this.DeleteByGroupAsync(groupName, name);
+        }
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/Domain/InterfaceImpl/VirtualMachineCustomImagesImpl.cs]: you should pass cancellationToken to this method

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/Domain/InterfaceImpl/VirtualMachineScaleSetsImpl.cs [hunk 14]
```diff
@@ -128,11 +128,10 @@ void Microsoft.Azure.Management.Compute.Fluent.IVirtualMachineScaleSets.Restart(
...
         /// <param name="groupName">The group the resource is part of.</param>
         /// <param name="name">The name of the resource.</param>
-        /// <return>An observable to the request.</return>
+        /// <return>A completable indicates completion or exception of the request.</return>
         async Task Microsoft.Azure.Management.Resource.Fluent.Core.CollectionActions.ISupportsDeletingByGroup.DeleteByGroupAsync(string groupName, string name, CancellationToken cancellationToken)
         {
- 
-            await this.DeleteByGroupAsync(groupName, name, cancellationToken);
+            await this.DeleteByGroupAsync(groupName, name);
         }
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/Domain/InterfaceImpl/VirtualMachineScaleSetsImpl.cs]: Why did you remove cancellationToken in this method call?

**Comment by hovsepm** [comment]
@anudeepsharma please take a look to my comments. Done with reviewing...

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/Common/TestHelper.cs (hunk 8 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/Common/TestHelper.cs]: Fixing tests and implementing this, forgot to put comment on this in first PR. Now fixed.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs (hunk 10 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/AvailabilitySetImpl.cs]: Fixed
